src/util/rest/mapping/prune.js

```diff
@@ -0,0 +1,24 @@
+const assert = require('assert');
+
+const getIndexVersions = async (call, idx) => {
+  const result = await call('GET', '', {
+    endpoint: `_cat/indices/${idx}@*`
+  });
+  return result.body.map(({ index }) => index.split('@')[1]);
+};
+
+const deleteIndexVersion = (call, idx, version) => call('DELETE', `${idx}@${version}`)
+  .then((r) => r.statusCode === 200 && r.body.acknowledged === true);
+
+module.exports = async (call, versions, idx) => {
+  const remoteVersions = await getIndexVersions(call, idx);
+  const localVersions = versions.get(idx);
+  assert(localVersions !== undefined, 'Index must be loaded');
+  const persistedVersions = Object.keys(localVersions);
```

Comment:
```suggestion
  const localVersions = Object.keys(versions.get(idx));
```
and raise in versions.get